src/tsolvers/egraph/Enode.h

```diff
@@ -277,6 +277,20 @@ class EnodeAllocator : public RegionAllocator<uint32_t>
 
 };
 
+
+/**
+ * Expl is a generalization of an explanation inequality that covers explicitly
+ * the standard inequality consisting of a negated equality or inequality,
+ * difference of two constants, and a boolean atom and its negation being distinct.
+ */
+struct Expl {
+    enum class Type { std, cons, pol, undef } type;
+    PtAsgn pta;
+    PTRef pol_term;
+    Expl() : type(Type::undef), pta(PtAsgn_Undef), pol_term(PTRef_Undef) {}
```

Comment:
@aehyvari , you mentioned in the description that you think this struct should be smaller? I am not sure how to make it one-word size, but it seems to me that `pta` member is valid only for type `std` and `pol_term` is valid only for `pol`, right? Maybe they can be put into a `union`?

Reply:
Yes.  I was thinking of packing the information in `Expl::Type` to the bits of `PtAsgn`, and packing `PtAsgn` so that two bits are reserved for the polarity (`l_True`, `l_False`, `l_Undef`).  But since there is no effect on performance in my plot, I'm planning to postpone this to a different PR.

Reply:
OK, makes sense.